File: java/src/LC_26_RemoveDuplicates.java

import java.util.Arrays;

// https://leetcode.com/problems/remove-duplicates-from-sorted-array/

public class LC_26_RemoveDuplicates {
    public int removeDuplicates(int[] nums) {
        int index = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[i - 1]) {
                nums[index++] = nums[i];
            }
        }
        return index;
    }

    public static void main(String[] args) {
        LC_26_RemoveDuplicates duplicates = new LC_26_RemoveDuplicates();
        int[] nums = { 1, 1, 2, 3, 3, 4, 5, 6, 6 };
        int index = duplicates.removeDuplicates(nums);
        System.out.println(Arrays.toString(Arrays.copyOf(nums, index)));
    }
}

File: python/LC_26_RemoveDuplicates.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]:
                nums[k] = nums[i]
                k += 1
        return k
    
sol = Solution()
nums = [1,1, 2, 3, 4, 5, 6, 6]
sol.removeDuplicates(nums)
print(nums)

nums = [1, 2, 3, 4, 5, 6, 7]
sol.removeDuplicates(nums)
print(nums)